darts/utils/timeseries_generation.py

```diff
@@ -256,6 +294,11 @@ def datetime_attribute_timeseries(time_index: Union[pd.DatetimeIndex, TimeSeries
     one_hot
         Boolean value indicating whether to add the specified attribute as a one hot encoding
         (results in more columns).
+    cyclic
+        Boolean value indicating whether to add the specified attribute as a cyclic encoding.
+        (adds 2 columns, corresponding to sin and cos transformation)
+    until
+        Extend the time_index into the future either by 'int' time points or until timestamp(`str` or `pd.Timestamp`).
```

Comment:
This could be somewhat misleading if the series is integer-indexed.
In this case the `int` could also be interpreted as the last time step we want data for and the behaviour wouldn't match that of the time-indexed series. You could maybe add two parameters (one of which only could be defined): `until: Option[Union[int, str, pd.Timestamp]]` would always define the last timestamp and `add_length: Option[int]` would define an additional number of steps. WDYT?

Reply:
Now that I think of it, it makes sense to separate these two behaviours into two different parameters.